src/feedspora/feedspora_runner.py

```diff
@@ -330,6 +292,108 @@ def test_output(self, text):
 
         return self.output_test(output)
 
+    # pylint: disable=too-many-branches
+    def shorten_url(self, the_url):
+        '''
+        Apply configured URL shortener (if present) to the provided link and
+        return the result.  If anything goes awry, return the unmodified link.
+        '''
+        to_return = the_url
+        # Default
+        short_options = {'timeout': 3}
+        short_options.update(self._url_shortener_opts)
+        if the_url and self._url_shortener and (self._url_shortener != 'none'):
+            try:
+                shortener = pyshorteners.Shortener(**short_options)
```

Comment:
Could this be simplified/refactored with:

```
to_return = the_url
try:
    shortener = pyshorteners.Shortener(**short_options)
    assert self._url_shortener in shortener.available_shorteners  # just to make sure the user cannot inject sth insecure
    to_return = getattr(pyshorteners, self._url_shortener)
except AssertionError:
    all_shorteners = " ".join(shortener.available_shorteners)
    logging.error("URL shortener %s is unimplemented!", self._url_shortener)
    logging.info("Available URL shorteners: %s", all_shorteners)
except AttributeError:
    pass
```

(not tested)

Reply:
I think I see your idea here, and it's worth checking out.  I'll investigate further...

Reply:
It doesn't work, unfortunately.  It throws the exception "module 'pyshorteners' has no attribute 'tinyurl'" (or whatever shortener you tried).  Unfortunately, it doesn't look like there is a way to do this using getattr.
I found another way to do this, however:
```
            try:
                shortener = pyshorteners.Shortener(**short_options)
                redirect = {
                    'adfly': shortener.adfly.short,
                    'bitly': shortener.bitly.short,
                    'chilpit': shortener.chilpit.short,
                    'clckru': shortener.clckru.short,
                    'dagd': shortener.dagd.short,
                    'isgd': shortener.isgd.short,
                    'osdb': shortener.osdb.short,
                    'owly': shortener.owly.short,
                    'post': shortener.post.short,
                    'qpsru': shortener.qpsru.short,
                    'soogd': shortener.soogd.short,
                    'tinycc': shortener.tinycc.short,
                    'tinyurl': shortener.tinyurl.short,
                    }
                if self._url_shortener in redirect:
                    to_return = redirect[self._url_shortener](the_url)
                    # Sanity check!
                    if len(to_return) > len(the_url):
                        # Not shorter?  You're fired!
                        raise RuntimeError('Shortener %s produced a longer ' +
                                           'URL than the original!',
                                           self._url_shortener)
                else:
                    all_shorteners = ' '.join(shortener.available_shorteners)
                    logging.error('URL shortener %s is unimplemented!',
                                  self._url_shortener)
                    logging.info('Available URL shorteners: %s',
                                 all_shorteners)
            # pylint: disable=broad-except
            except Exception as exception:
                # Shortening attempt failed - revert to non-shortened link
                logging.error('Cannot shorten URL %s with %s: %s',
                              the_url, self._url_shortener, str(exception))
                to_return = the_url
            # pylint: enable=broad-except
```
This "works", except for one nasty problem I found:  if any of the redirect values are (or even _become_) invalid, **every other shortener function then fails**, since the redirect dict is broken.  Worse, at least in some ways, the error that's thrown is indicating the broken shortener, not the one you're trying to use.  It took me a while to figure out that problem, and what was causing it.
All in all, I think the multiple branch approach is "safer" even if it's uglier.  And it leads to less brittle code, ultimately.
...but I'd be happy to revisit this issue should I become a more ninja-like Pythonista some day!

Reply:
I'm not sure I understand the problem you describe.

The attribute error calling `getattr` on `shortener` is raised with `pyshortener==0.6.1` (installed from pypi), but not with `pyshortener==1.0.0` (installed from git with `pip install git+https://github.com/ellisonleao/pyshorteners.git`).  The following gist works on my machine : https://gist.github.com/aurelg/e7c5b1f52a754ed00eebadadee9382f6

Reply:
Thank you so much for your example!  As it turns out, the problem I was having was caused by running getattr against the generic pyshorteners rather than against the shortener instance.  Once I understood that (duh!), I was able to implement the solution using getattr, for much better code sustainability (as well as getting rid of all those nasty branches and the pylint disabler).  I've pushed that code now, so we're in business!